File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool palindromeString(int i, string &s)
    {
        int n = s.size();
        if(i >= n/2) return true;
        if(s[i] != s[n-i-1]) return false;
        return palindromeString(i+1, s);
    }

    bool isPalindrome(string s) {
        string filterString = "";
        for(char c: s)
        {
            if(isalnum(c))
            {
                filterString += tolower(c);
            }
        }
        return palindromeString(0,filterString);
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        st, end = 0, len(s) - 1

        while st < end:
            while st < end and not s[st].isalnum():
                st += 1
            while st < end and not s[end].isalnum():
                end -= 1
            
            if s[st].lower() != s[end].lower():
                return False
            
            st += 1
            end -= 1
        return True